### ISSUE TITLE
insecure registry and list restricted to first 100 entries

### ISSUE THREAD

**Issue by vikramkhatri** [byrnedo/docker-reg-tool#4]
Thank you for the script. Very useful.

I needed to add `-k` in `curl` as my local registry is insecure.

Any way - the `list` reports only 100 entries. This may be restriction from the API. Any idea on getting list of all entries. My private registry has more than 300 entries.

**Comment by byrnedo** [comment]
Hi @vikramkhatri  I'll have a look at adding the insecure flag. 
Regarding the paging I'll have a look

**Comment by byrnedo** [comment]
Added `INSECURE_REGISTRY` env flag in 703f90f254f03bbc9eb49c53fe720916b2a1f6d0

**Comment by byrnedo** [comment]
Had a quick look at https://docs.docker.com/registry/spec/api/#listing-image-tags
I tried to set the query string to the parameters they name but it didn't seem to affect the result.
It could be that the registry I tested against is too old..

**Comment by pmossakx** [comment]
@vikramkhatri You can set it inside case for `list`:
```
git diff
diff --git a/docker_reg_tool b/docker_reg_tool
index 8b5857e..edc43a0 100755
--- a/docker_reg_tool
+++ b/docker_reg_tool
@@ -101,7 +101,7 @@ case "$ACTION" in
                 curlCmd -s "$PROTO$REG/v2/$repo/tags/list" | jq -r '.tags|.[]'
             fi
         else
-            curlCmd -s "$PROTO$REG/v2/_catalog" | jq -r '.repositories|.[]'
+            curlCmd -s "$PROTO$REG/v2/_catalog?n=500" | jq -r '.repositories|.[]'
         fi
```
Now it lists all repos:
`docker_reg_tool https://mydockerregistry.com list | wc -l`
`328`

**Comment by byrnedo** [comment]
Implemented in 9dc8e79.
